schema-lending.graphql

```diff
@@ -77,203 +145,273 @@ interface Protocol {
+  " Revenue claimed by suppliers to the protocol. LPs on DEXs (e.g. 0.25% of the swap fee in Sushiswap). Depositors on Lending Protocols. NFT sellers on OpenSea. "
+  cumulativeSupplySideRevenueUSD: BigDecimal!
 
-  usageMetrics: [UsageMetricsDailySnapshot!]! @derivedFrom(field: "protocol")
+  " Gross revenue for the protocol (revenue claimed by protocol). Examples: AMM protocol fee (Sushi’s 0.05%). OpenSea 10% sell fee. "
+  dailyProtocolSideRevenueUSD: BigDecimal!
 
-  financialMetrics: [FinancialsDailySnapshot!]! @derivedFrom(field: "protocol")
+  " Gross revenue for the protocol (revenue claimed by protocol). Examples: AMM protocol fee (Sushi’s 0.05%). OpenSea 10% sell fee. "
+  cumulativeProtocolSideRevenueUSD: BigDecimal!
 
-  ##### Lending-Specific #####
+  " All revenue generated by the protocol. e.g. 0.30% of swap fee in Sushiswap, all yield generated by Yearn. "
+  dailyTotalRevenueUSD: BigDecimal!
 
-  " All markets that belong to this protocol "
-  markets: [Market!]! @derivedFrom(field: "protocol")
+  " All revenue generated by the protocol. e.g. 0.30% of swap fee in Sushiswap, all yield generated by Yearn. "
+  cumulativeTotalRevenueUSD: BigDecimal!
 
-  lendingType: LendingType
+  " Total assets deposited on a given day, in USD "
+  dailyDepositUSD: BigDecimal!
 
-  riskType: RiskType
+  " Sum of all historical deposits in USD (only considers deposits and not withdrawals) "
+  cumulativeDepositUSD: BigDecimal!
+
+  " Total assets borrowed/minted on a given day, in USD. "
+  dailyBorrowUSD: BigDecimal
+
+  " Sum of all historical borrows/mints in USD (i.e. total loan origination). "
+  cumulativeBorrowUSD: BigDecimal
+
+  " Total supply of minted tokens in native amounts. Only applies to CDP "
+  mintedTokenSupplies: [BigInt!]
+  
+  " Block number of this snapshot "
+  blockNumber: BigInt!
+
+  " Timestamp of this snapshot "
+  timestamp: BigInt!
 }
 
 ###############################
 ##### Pool-Level Metadata #####
 ###############################
 
-
 type Market @entity {
   " Smart contract address of the market "
   id: ID!
 
   " The protocol this pool belongs to "
   protocol: LendingProtocol!
 
-  # Generally protocols accept one or multiple tokens and mint tokens to the depositor to track ownership
-  # Some protocols reward DAO tokens or other incentivisation tokens to holders of LP tokens
-  # Some protocols don't mint any tokens to track ownership, in that case outputToken is null 
-  # and inputToken balances are used to calculate returns
+  " Token that need to be deposited to take a position in protocol "
+  inputToken: Token!
 
-  " Tokens that need to be deposited to take a position in protocol. e.g. WETH and USDC to deposit into the WETH-USDC pool "
-  inputTokens: [Token!]!
-
-  " Token that is minted to track ownership of position in protocol "
+  " Token that is minted to track ownership of position in protocol (e.g. aToken, cToken). Leave as null if doesn't exist "
   outputToken: Token
```

Comment:
have we found a lending protocol that does not have outputToken? if not, maybe leave it required?

(if this makes sense to you, also change exchangeRate to required)

Reply:
yeah, none of the CDPs have outputToken